### PR TITLE
Componentizes small cells

### PR DESCRIPTION
**Pull request by Tarmunora**
## About the PR 
Handling for small cells (used in guns/etc) and things that use them (guns, batons, etc) is done via components
Also adds a few new signals, and a wrapper proc for attackby so that signals work right with it

Also makes SEND_SIGNAL do nothing when trying to signal `null` instead of runtiming

## Why's this needed? 
Consistency in things that use small power cells.

### CODE REVIEW

**Comment by Tarmunora** [comment]
oops forgot to clean some stuff up

File: code/atom.dm [hunk 1]
```diff
@@ -632,10 +632,16 @@
...
+/atom/proc/Attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
+	SHOULD_NOT_OVERRIDE(1)
 	if(SEND_SIGNAL(src,COMSIG_ATTACKBY,W,user))
 		return
+	src.attackby(W, user, params, is_special)
+
+//mbc : sorry, i added a 'is_special' arg to this proc to avoid race conditions.
+///internal proc for when an atom is attacked by an item. Override this, but do not call it,
+/atom/proc/attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
 	src.material?.triggerOnHit(src, W, user, 1)
```
Comment (PrimeNumb) [inline code/atom.dm:645]:
```suggestion
/atom/proc/attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
	PROTECTED_PROC(TRUE)
	src.material?.triggerOnHit(src, W, user, 1)
```
possibly? won't prevent people from calling it from within the type but will provide some level of protection at the very least

Comment (PrimeNumb) [inline code/atom.dm:645]:
```suggestion
/atom/proc/attackby(obj/item/W as obj, mob/user as mob, params, is_special = 0)
	PROTECTED_PROC(TRUE)
	src.material?.triggerOnHit(src, W, user, 1)
```
possibly? won't prevent calls from atom subtypes but it'd yell if datums did (i.e. item specials etc)

Comment (Tarmunora) [reply code/atom.dm:645]: Maybe

**Comment by Tarmunora** [comment]
yolo

**Pull request merged**